Change test file names to avoid conflicts when running tests

bobsayshilol opened the pull request:
`ctest -j <N>` can fail sporadically due to tests sharing the same temporary file names. This PR fixes the issue by changing the names of all those that conflict.

I'm not sure if `get_unique_test_name()` is the best approach, but it was simpler than changing every string.

evpobr commented:
Autotools build fails:

```
ERROR: files left in build directory after distclean:
./lct_opus.opus
make[1]: *** [distcleancheck] Error 1
Makefile:3429: recipe for target 'distcleancheck' failed
make[1]: Leaving directory '/home/runner/work/libsndfile/libsndfile/libsndfile-1.0.31/_build/sub'
Makefile:3356: recipe for target 'distcheck' failed
make: *** [distcheck] Error 1
Error: Process completed with exit code 2.
```

bobsayshilol commented:
> Autotools build fails:

Yeah, and looking through the logs I see `opus.opus` appear more than once which my script didn't catch from a test run since I wasn't building with that enabled locally, so more fixes are required there too.

I had a brief look at the autotools issue but I don't know enough about autotools to pinpoint where it decides what to cleanup, so can you point me in the right direction please?

evpobr commented:
For Autotools ask @SoapGentoo :smile:

SoapGentoo commented:
@bobsayshilol I'll have a look

SoapGentoo commented:
@bobsayshilol could you squash and rebase on latest `master`? I can't reproduce the CI issue locally

bobsayshilol commented:
> @bobsayshilol could you squash and rebase on latest `master`? I can't reproduce the CI issue locally

I've fixed up the remaining filename collisions, and then rebased and squashed onto master but it looks like it's still failing:
```
 ERROR: files left in build directory after distclean:
./lct_opus.opus
make[1]: *** [distcleancheck] Error 1
```

It's strange that `macos-autotools` is passing though since presumably it's also using automake?

SoapGentoo commented:
@bobsayshilol can I push to your branch? I have a hunch

bobsayshilol commented:
Sure, go for it.

SoapGentoo commented on tests/utils.tpl:
```diff
@@ -851,6 +853,15 @@ check_open_file_count_or_die (int lineno)
+get_unique_test_name (const char ** filename, const char * test)
+{	static char	buffer [1024] ;
+
+	snprintf (buffer, sizeof (buffer) - 1, "%s_%s", test, *filename) ;
```
@bobsayshilol foudn the bug. For the final review, could you amend this? `snprintf` assumes the maximum length it can use is `n-1`, i.e. it's already `NUL`-termination aware and doesn't require passing `n-1` into it:

```suggestion
	snprintf (buffer, sizeof (buffer), "%s_%s", test, *filename) ;
```

bobsayshilol replied:
Derp. Good spot!

bobsayshilol commented:
Nice catch @SoapGentoo!

SoapGentoo commented on tests/compression_size_test.c:
```diff
@@ -37,6 +37,8 @@
 #define	SAMPLE_RATE		16000
 #define	DATA_LENGTH		(SAMPLE_RATE)
 
+#define	CMP_TEST_PREFIX	"cmp"
```
Let's move away from macros, which are ugly. Let's go with [Ulrich Drepper's suggestion](https://udrepper.livejournal.com/15119.html):
```suggestion
static const char CMP_TEST_PREFIX[] = "cmp";
```

bobsayshilol replied:
Ah that was a leftover from my original approach of prepending `*_TEST_PREFIX` to every filename, which quickly got out of hand. I'll change them over.

SoapGentoo reviewed (approved):
Thanks, I'm happy now, @evpobr please check too
